application.py

```diff
@@ -73,6 +73,12 @@ def login():
         return render_template("registration.html", message="you haven't registered please register first")
 
 
+@app.route("/bookpage/<string:isbn>", methods = ["GET"])
+def bookpage(isbn):
+    all_book = Books.query.filter(Books.isbn == isbn).first_or_404()
+    print(all_book.isbn, all_book.title, all_book.author, all_book.year, session["username"])
+    return render_template("bookpage.html", isbn = all_book.isbn, title = all_book.title, author = all_book.author, year = all_book.year, user = session["username"])
```

Comment:
send the whole object instead of sending each value as a parameter

Reply:
Yes mam